Initial support for remote debugging

drakhaevn opened the pull request:
Added initial bindings for connecting to a remote debug server. Still considered WIP + only TCP connection support is available at this time.

drakhaevn committed:
Initial support for remote debugging

Added initial bindings for connecting to a remote debug server. Still considered WIP + only TCP connection support is available at this time.

drakhaevn committed:
Merge branch 'iilegacyyii:dev' into dev

iilegacyyii commented:
Nice! Gave it a quick read and looks solid so far. Will do some testing on my end and will get back to you tomorrow on what next steps are with this.

iilegacyyii committed:
Fixed symbol resolution for remote targets

Previously we were initializing DbgHelp in haphazardly using a duplicated handle of the debuggee. This of course did not work for remote targets, and so has been replaced with a call to `IDebugSystemObjects4->GetCurrentProcessHandle`, which returns a handle dedicated to DbgHelp symbol management.

iilegacyyii commented:
Fixed the symbol resolution - DbgHelp was being initialised using a handle that would be only available on a local target, as such has been reworked. Check out 46bf3a741bfc3bbff2505516a6a876f0b88e4473 for more info
